refactor!: dropping pagination from getLogsByTags

benesjan opened the pull request:
In this PR I drop pagination from `getLogsByTags` endpoint because pagination here doesn't really make sense because we get more logs per tag only if there are multiple devices sending logs from a given sender to a recipient.

This is the final of 3 PRs in which I clean up the `getLogsByTags` endpoint:
1. First PR - including block timestamp in return value,
2. This is the second PR - I type the tag arg of the function to be `SiloedTag`,
3. in the last PR I will drop pagination from this endpoint.

# Potential DoS vector

Log tag is unconstrained and hence can be arbitrarily chosen by the sender. This introduces a potential DoS vector because the node currently loads all the logs into memory and returns them. Hence it could be feasible to re-use the same tag thousands of time and then spam a node by request for this tag.

I think the solution here is to simply define the maximum number of logs a node is willing to store per tag in the db. Given that there is no legitimate use case for log reuse I think making it even as low as 5 would be fine.

Note that this PR didn't introduce this DoS vector because even though I dropped the pagination here it didn't seem to be enforced before (if `limitPerTag` arg was undefined we just returned it all). For this reason I think we can merge it and tackle the issue in a followup PR.

benesjan commented:
* **#19290** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19290?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19289** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19289?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19233** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19233?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19125** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19125?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19161** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19161?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19161?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#19158** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19158?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19157** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19157?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#19030** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19030?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17611** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17611?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17568** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17568?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17534** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17534?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17445** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17445?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* **#17451** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/17451?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

spalladino commented:
> Note that this PR didn't introduce this DoS vector because even though I dropped the pagination here it didn't seem to be enforced before (if limitPerTag arg was undefined we just returned it all). For this reason I think we can merge it and tackle the issue in a followup PR.

Heads up that's not correct. The limit was enforced at the schema level (maybe not the wisest decision in retrospective, since it's unclear from the code), where we'd default the `limitPerTag` to the max value if not set. Still, I'm fine if you want to just remove it, as long as you add some guardrails in a future PR.

benesjan committed:
refactor!: dropping pagination from getLogsByTags

In this PR I drop pagination from `getLogsByTags` endpoint because pagination here doesn't really make sense because we get more logs per tag only if there are multiple devices sending logs from a given sender to a recipient.

This is the final of 3 PRs in which I clean up the `getLogsByTags` endpoint:
1. First PR - including block timestamp in return value,
2. This is the second PR - I type the tag arg of the function to be `SiloedTag`,
3. in the last PR I will drop pagination from this endpoint.

# Potential DoS vector

Log tag is unconstrained and hence can be arbitrarily chosen by the sender. This introduces a potential DoS vector because the node currently loads all the logs into memory and returns them. Hence it could be feasible to re-use the same tag thousands of time and then spam a node by request for this tag.

I think the solution here is to simply define the maximum number of logs a node is willing to store per tag in the db. Given that there is no legitimate use case for log reuse I think making it even as low as 5 would be fine.

Note that this PR didn't introduce this DoS vector because even though I dropped the pagination here it didn't seem to be enforced before (if `limitPerTag` arg was undefined we just returned it all). For this reason I think we can merge it and tackle the issue in a followup PR.

benesjan commented:
Pagination got re-introduced in https://github.com/AztecProtocol/aztec-packages/pull/19776
